Fixed colors for merchants

tomschw opened the pull request:
This PR will sort the lines and legends in the graphs in lexicographical order and the merchants now have a self-defined line color.

Commit:
Add more colours and reserve red and green for two-bound and cheapest merchant.

Commit:
Color is set by merchant.

Commit:
Fix total revenue share error where the plot would have too many points.

Commit:
Add sorting to the legend based on name.

Bouncner commented on asset/js/app.js:
```diff
@@ -249,6 +249,9 @@
                 return ((merchant_id in merchants) && merchants[merchant_id].merchant_name) ? merchants[merchant_id].merchant_name : merchant_id.substring(0, 8);
             },
+            getMerchantColor: function(merchant_id) {
+                return (merchant_id in merchants) ? merchants[merchant_id].color : '#FFFFFF';
```
Why white? That's pretty much transparent, isn't it?

tomschw replied:
It is basically just a fail safe. This only happens if we have a merchant which the marketplace doesn't know, which should never happen.

tomschw replied:
@Bouncner Fixed issue.

Bouncner commented on asset/js/app.js:
```diff
@@ -500,34 +506,36 @@
                     });
 
-                    sortLegend(chart);
+                    // sortLegend(chart);
```
?

Bouncner commented on asset/js/app.js:
```diff
@@ -573,7 +582,27 @@
-        function addPointToLine(chart, point, lineID, lineName, stepEnabled, lineToUse) {
+        function sortLegend(chart) {
+            var line;
+
```
Too much whitespace.

Bouncner commented on asset/js/app.js:
```diff
@@ -942,7 +906,7 @@
                         text: 'Date'
                     },
-                    showEmpty: false
+                    showEmpty: false,
```
?

Commit:
Address review.

tomschw committed:
Update app.js

Add thrown error when merchant_id is not in our merchant list. Also color these merchant_ids in light grey.

Commit:
Merge branch 'fixed_colors_for_merchants' of github.com:hpi-epic/pricewars-mgmt-ui into fixed_colors_for_merchants

cmfcmf commented:
> This PR will sort the lines and legends in the graphs in lexicographical order and the merchants now have a self-defined line color.

That's so great – the constantly changing colors were so confusing!

Bouncner commented on asset/js/app.js:
```diff
@@ -250,7 +250,11 @@
+                    return merchants[merchant_id].color;
+                }
+                console.error("Found an merchant id that is not included in our merchants from the marketplace.");
+                return '#E1E1E1';
```
"included in our merchants from the marketplace"? Please rewrite.\
How did you "find" a merchant id?

Also: short comment which color this is.

tomschw replied:
Is it better now?

tomschw committed:
Update app.js

Edit error.
